exist-core/src/main/java/org/exist/xquery/functions/fn/FunCollationKey.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * eXist-db Open Source Native XML Database
+ * Copyright (C) 2001 The eXist-db Authors
+ *
+ * info@exist-db.org
+ * http://www.exist-db.org
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package org.exist.xquery.functions.fn;
+
+import org.apache.commons.codec.binary.Base64;
+import org.exist.dom.QName;
+import org.exist.util.Collations;
+import org.exist.xquery.*;
+import org.exist.xquery.value.*;
+
+import com.ibm.icu.text.Collator;
+
+import java.nio.charset.StandardCharsets;
+
+public class FunCollationKey extends BasicFunction {
+
+    private static final QName FN_NAME = new QName("collation-key", Function.BUILTIN_FUNCTION_NS, FnModule.PREFIX);
```

Comment:
Can we use the `FunctionDSL` for the function signatures please? If you look for uses of `FunctionDSL` in the `fn` module you should find some existing examples